GSA/searchgov-spider
# SRCH-5430 full crawl testing

## Summary
**Changes**
- This PR removes `betamax` and uses built in `scrapy` cache functionalty to cache full crawls of both spiders.
- Adds a `full_crawl_test.py` file that tests full crawls of the test quote sites once with `domain_spider` and once with `domain_spider_js`.
- The cached files are in `tests/search_gov_spiders/scrapy_httpcache`.  This format is the only one I could find that would work locally AND in circle.  I spend a long time trying many things...
 - I upgraded us back to python 3.12 since everything seemed to work in circle for me.  Added the branch name and version to the cache key to help with cache errors.
 - Minor changes to the pipelines file because i could not even import it without an error opening the `all-links.csv` file.  Moved it from a class variable to inside `process_item`.
 - Removed old tests and cache files.

**How to Test**
- Run `pytest tests` to confirm tests work locally with cache.
- Disable internet and run `pytest tests` again.  Tests should still pass.
- If you want, delete contents of httpcache directory and rerun tests.  New files should be created -- and it should take a lot longer.

### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
- [X] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures:  #### Process Checks
